design/product-design-list-all-accounts.md

```diff
@@ -0,0 +1,75 @@
+# LIST AND CREATE ACCOUNTS
+
+## List accounts
+
+Shows all accounts in system, has a button that allows to go to the account creation view.
+The name of each acount can be clicked and this will lead to a blank page (for the moment)
+
+## Create account
+
+Shows a form to enter the account information:
+
+- Account name: Must be unique among all other names on the system.
+- Person fullname: should be minimum 1 character long or maximun 255 characters.
+
+Finally when we press submit button, if the content provided in the form is valid, this will redirect to list accounts page, else will show an feedback error message showing form mistakes.
+
+## Wireframes
+
+Images (mocks)
+
+- List accounts page
+- Forms page
+- Forms page error
+
+## API contract
+
+### List accounts page
+
+| RES API spec  | content       |
+| ------------- |:-------------:|
+| Request headers     | Content-Type: application/json |
+| Request HTTP method | GET      |
+| Request URL       | /account/list      |
+| Request body       | ---      |
+| Response headers       | Content-Type: application/json     |
+| Response body       | {"data": [{account: "username1", "fullname": "User Name 1",...}]      |
+| Response status code       | 200     |
+
+### Forms page
+
+#### Showing form
+
+| RES API spec  | content       |
+| ------------- |:-------------:|
+| Request headers     | Content-Type: text/html; charset=utf-8 |
+| Request HTTP method | GET      |
+| Request URL       | /account/new      |
+| Request body       | ---      |
+| Response headers    | Content-Type: text/html; charset=utf-8     |
+| Response body       | `<html>...</html>`      |
```

Comment:
REST APIs return data (JSON), not HTML. The client (SPA) is solely responsible for rendering the HTML.
If it helps in thinking about this API/client split, imagine that we're going to build a mobile app in the future, and we want to re-use the API. Thus; it can only return data.

Applies throughout.